packages/mongo-nav/src/org-nav/OrgNav.tsx

```diff
@@ -304,12 +340,40 @@ export default function OrgNav({
           </OrgNavLink>
         )}
 
-        <UserMenu
-          account={account}
-          activeProduct={activeProduct}
-          urls={urls}
-          hosts={hosts}
-        />
+        {isOnPrem ? (
+          <div className={onPremMenuWrapper}>
+            <UserMenuTrigger
+              name={account.firstName}
+              open={onPremMenuOpen}
+              setOpen={setOnPremMenuOpen}
+            />
+            <Menu open={onPremMenuOpen} setOpen={setOnPremMenuOpen}>
+              <MenuItem href={urls.onPrem.profile}>Profile</MenuItem>
+              <MenuItem href={urls.onPrem.mfa}>
+                Two-factor Authentication
+              </MenuItem>
+              <MenuItem href={urls.onPrem.personalization}>
+                Personalization
+              </MenuItem>
+              <MenuItem href={urls.onPrem.invitations}>Invitations</MenuItem>
+              <MenuItem href={urls.onPrem.organizations}>
+                Organizations
+              </MenuItem>
+              <MenuItem href={urls.onPrem.featureRequest}>
+                Feature Request
+              </MenuItem>
+              <MenuItem onClick={onLogout}>Sign Out</MenuItem>
```

Comment:
This is going to be a major one where we'll want a `data-testid` to be assigned, because we click this many times in the E2E tests. The same should already exist (or be added) to the UserMenu.

I would actually go as far as to say it would be nice to have testids for each of the links, plus the trigger, for both menus. That's a fair number of similar testids, so to quote PEP20 (The Zen of Python):
> "Namespaces are one honking great idea—let's do more of those!"

(aka I hadn't thought this through, but it'd be nice to use prefixes like `om-user-menu` and `user-menu` to keep things tidy)

...also, I am so glad I didn't get PEP20 tattooed on my arm when I was going through my Python phase 😬

Reply:
👍